Tasks/AzureRmWebAppDeployment/Strings/resources.resjson/en-US/resources.resjson

```diff
@@ -33,11 +34,22 @@
   "loc.input.help.AdditionalArguments": "Additional Web Deploy arguments that will be applied when deploying the Azure Web App like,-disableLink:AppPoolExtension -disableLink:ContentExtension.",
   "loc.input.label.TakeAppOfflineFlag": "Take App Offline",
   "loc.input.help.TakeAppOfflineFlag": "Select the option to take the AzureRM Web App offline by placing an app_offline.htm file in the root directory of the Web App before the sync operation begins. The file will be removed after the sync operation completes successfully.",
+  "loc.input.label.XmlTransformsAndVariableSubstitutions": "XML Transforms & Variable Substitutions",
+  "loc.input.help.XmlTransformsAndVariableSubstitutions": "Select to enable xml transforms",
+  "loc.input.label.XdtTransformation": "Transformation",
+  "loc.input.help.XdtTransformation": "The config transfoms will be run for `*.Release.config` and `*.<EnvironmentName>.config` on the `*.config file`. Config transforms will be run prior to the Variable Substitution.",
+  "loc.input.label.VariableSubstitution": "Variable Substitution",
+  "loc.input.help.VariableSubstitution": "Variables defined in the Build or Release Definition will be automatically substituted in the appSettings, applicationSettings, and connectionStrings section of the config files. If same variables are defined in the Release Definition and in the Environment, then the Environment variables will supersede the Release Definition variables. Variable Substitution is run after Config Transforms.",
+  "loc.input.label.JSONVariableSubstitutionsFlag": "JSON Variable Substitutions",
+  "loc.input.help.JSONVariableSubstitutionsFlag": "Enable this option for JSON variable Substitution",
+  "loc.input.label.JSONVariableSubstitutions": "Variable Substitution",
+  "loc.input.help.JSONVariableSubstitutions": "Provide a line separated list of JSON files for substituting variables in them. The files names are relative to the root folder. For substituting JSON variables that are in a hierarchy, specify them using the format Data.DefaultConnection.ConnectionString in the Environment/Definition variable. If same variables are defined in the Definition and Environment, then the Environment variables will supersede the Definition variables. Variable Substitution is run after Config Transforms. Build/Release/System defintion variables are eliminated in substitution.",
```

Comment:
Provide new line separated list of JSON files to substitute the variable values in. Files names are to be provided relative to the root folder. 
To substitute JSON variables that are nested or hierarchical, specify them using JSONPath expressions. 
For example, to replace the value of ‘ConnectionString’ in the sample below, you need to define a variable as ‘Data.DefaultConnection.ConnectionString’ in the build/release definition (or release definition’s environment). <br/>
```
{
    "Data": {
      "DefaultConnection": {
        "ConnectionString": "Server=(localdb)\\SQLEXPRESS;Database=MyDB;Trusted_Connection=True"
      }
    }  
}
```
Variable Substitution is run after configuration transforms. Build/Release’s system definition variables are excluded in substitution.